test/gui/shared/steps/server_context.py

```diff
@@ -230,3 +230,28 @@ def step(context, user):
         data['password'],
         data['expireDate'],
     )
+
+
+@Then('user "|any|" in the server should have a share with these details:')
+def step(context, user):
+    data = {
+        'share_type': None,
+        'uid_owner': None,
+        'permissions': None,
+        'path': None,
+        'item_type': None,
+        'share_with': None,
+    }
+    for row in context.table:
+        key, value = row
+        if key in data:
+            data[key] = value
+    sharing_helper.share_details(
+        user,
+        data['share_type'],
+        data['uid_owner'],
+        data['permissions'],
+        data['path'],
+        data['item_type'],
+        data['share_with'],
+    )
```

Comment:
```suggestion
        share = get_share(path, share_type, share_with)
        if key in data:
          # check that key:value exist in the share
```

Reply:
but to check this from here: we need `PERMISSIONS`  and `share_types`, if we import them, there will be a cyclic import